interfaces/acados_template/acados_template/acados_casadi_ocp_solver.py

```diff
@@ -489,23 +557,26 @@ def _get_cost_node(self, i, N_horizon, xtraj_node, utraj_node, ptraj_node, sl_no
                     ptraj_node[0],
                     sl_node[0],
                     su_node[0],
-                    ocp.get_initial_cost_expression(),
+                    self.get_cost_expression(ocp, ptraj_node[0][dims.np:dims.ny+dims.np], i),
+                    ca.vertcat(ocp.model.p, ptraj_node[0][dims.np:dims.np+dims.ny]),
                     dims.ns_0, cost.zl_0, cost.Zl_0, cost.zu_0, cost.Zu_0)
         elif i < N_horizon:
             return (xtraj_node[i],
                     utraj_node[i],
                     ptraj_node[i],
                     sl_node[i],
                     su_node[i],
-                    ocp.get_path_cost_expression(),
+                    self.get_cost_expression(ocp, ptraj_node[i][dims.np:dims.ny+dims.np], i),
+                    ca.vertcat(ocp.model.p, ptraj_node[i][dims.np:dims.np+dims.ny]),
                     dims.ns, cost.zl, cost.Zl, cost.zu, cost.Zu)
         else:
             return (xtraj_node[-1],
                     [],
                     ptraj_node[-1],
                     sl_node[-1],
                     su_node[-1],
-                    ocp.get_terminal_cost_expression(),
+                    self.get_cost_expression(ocp, ptraj_node[-1][dims.np:(dims.nx if dims.ny else 0)+dims.np], i),
+                    ca.vertcat(ocp.model.p, ptraj_node[-1][dims.np:(dims.nx if dims.ny else 0)+dims.np]),
```

Comment:
Also here. Plus, I don't know why there is an `nx`. I think it should only depend on `ny`. Am I missing something?